src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -1645,5 +1645,173 @@ internal static Exception CreateInvalidNameCharException(string name, int index,
             return CreateException(index == 0 ? SR.Xml_BadStartNameChar : SR.Xml_BadNameChar, XmlException.BuildCharExceptionArgs(name, index), exceptionType, 0, index + 1);
         }
+
+        internal static bool TryFormat(bool value, Span<char> destination, out int charsWritten)
+        {
+            string valueAsString = value ? "true" : "false";
+
+            charsWritten = valueAsString.Length;
+            return valueAsString.TryCopyTo(destination);
+        }
+
+        internal static bool TryFormat(char value, Span<char> destination, out int charsWritten)
+        {
+            charsWritten = 1;
+            if (destination.Length < 1) return false;
+
+            destination[0] = value;
+            return true;
+        }
+
+        internal static bool TryFormat(decimal value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, NumberFormatInfo.InvariantInfo);
+        }
+
+        internal static bool TryFormat(sbyte value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out  charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(short value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(int value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(long value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(byte value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(ushort value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(uint value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(ulong value, Span<char> destination, out int charsWritten)
+        {
+            return value.TryFormat(destination, out charsWritten, default, CultureInfo.InvariantCulture);
+        }
+
+        internal static bool TryFormat(float value, Span<char> destination, out int charsWritten)
+        {
+            ReadOnlySpan<char> valueSpan = default;
+
```

Comment:
How do we handle NaN? Are we relying on value.TryFormat to handle that case?

There are a few things that can be improved here. Take a look at [Number.FormatSingle](https://github.com/dotnet/runtime/blob/a272954f5c88544467112d944f662ef81b3d7487/src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs#L649-L657). You have doing 3 special case checks here for every number when it isn't a special case. FormatSingle uses float.IsFinite to split off the special case code into it's own condition, meaning most numbers will only have a single check before formatting.

Reply:
@mconnew  I received following feedback before : https://github.com/dotnet/runtime/pull/76436#discussion_r1011714502
 Maybe we can leave this optimization for another PR, as it will be good Format to string and to Span<char> to have similar code. Then we can improve both methods with this trick(Thanks for pointing it to me).

Reply:
I 100% agree with prioritizing readability as it doesn't matter how fast incorrect code runs. In this case I don't think it sacrifices readability. This is what I was suggesting:
```c#
        internal static bool TryFormat(float value, Span<char> destination, out int charsWritten)
        {
            ReadOnlySpan<char> valueSpan = default;

            if (!float.IsFinite(value))
            {
                if (float.IsNaN(value))
                    valueSpan = "NaN";
                else
                    valueSpan = float.IsNegative(value) ? "-INF" : "INF";
            }
            else if (IsNegativeZero((double)value))
            {
                valueSpan = "-0";
            }
            else
            {
                return value.TryFormat(destination, out charsWritten, "R", NumberFormatInfo.InvariantInfo);
            }

            charsWritten = valueSpan.Length;
            return valueSpan.TryCopyTo(destination);
        }
```

I don't think that complicates the code and is very easy to read. It reduces the number of if checks for a regular number from 4 down to 2.

Reply:
@mconnew Done + changed TryFormat for double too using the same trick